gcloud/monitoring/resource.py

```diff
@@ -0,0 +1,183 @@
+        self.description = description
+        self.labels = labels
+
+    @classmethod
+    def _fetch(cls, client, resource_type):
+        """Look up a resource descriptor by type.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type resource_type: string
+        :param resource_type: The resource type name.
+
+        :rtype: :class:`ResourceDescriptor`
+        :returns: The resource descriptor instance.
+
+        :raises: :class:`gcloud.exceptions.NotFound` if the resource descriptor
+            is not found.
+        """
+        path = '/projects/{project}/monitoredResourceDescriptors/{type}' \
+            .format(project=client.project,
+                    type=resource_type)
+        info = client.connection.api_request(method='GET', path=path)
+        return cls._from_dict(info)
+
+    @classmethod
+    def _list(cls, client, filter_=None):
+        """List all resource descriptors for the project.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type filter_: string or None
+        :param filter_: An optional filter string describing the resource
+            descriptors to be returned. See the `filter documentation`_.
+
+        :rtype: list of :class:`ResourceDescriptor`
+        :returns: A list of resource descriptor instances.
+
+        .. _filter documentation:
+            https://cloud.google.com/monitoring/api/v3/filters
+        """
+        path = '/projects/{project}/monitoredResourceDescriptors/'.format(
+            project=client.project)
+
+        descriptors = []
+
+        page_token = None
+        while True:
+            params = {}
+
+            if filter_ is not None:
+                params['filter'] = filter_
+
+            if page_token is not None:
+                params['pageToken'] = page_token
+
+            response = client.connection.api_request(
+                method='GET', path=path, query_params=params)
+            for info in response.get('resourceDescriptors', ()):
+                descriptors.append(cls._from_dict(info))
+
+            page_token = response.get('nextPageToken')
+            if not page_token:
+                break
+
+        return descriptors
+
+    @classmethod
+    def _from_dict(cls, info):
+        """Construct a resource descriptor from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`ResourceDescriptor`
+        :returns: A resource descriptor.
+        """
+        return cls(
+            name=info['name'],
+            type=info['type'],
+            display_name=info.get('displayName', ''),
+            description=info.get('description', ''),
+            labels=tuple(LabelDescriptor._from_dict(label)
+                         for label in info.get('labels', [])),
```

Comment:
Ditto here. Missed in https://github.com/GoogleCloudPlatform/gcloud-python/pull/1691/commits/41e8ff20c658ce1d29574a1770f20d11dc266188